app/controllers/images_controller.rb

```diff
@@ -68,80 +50,49 @@ def edit
   end
 
   def update
-    Edition.find_and_lock_current(document: params[:document]) do |edition| # rubocop:disable Metrics/BlockLength
-      image_revision = edition.image_revisions.find_by!(image_id: params[:image_id])
-      image_updater = Versioning::ImageRevisionUpdater.new(image_revision, current_user)
-
-      image_updater.assign(update_params)
-      next_image_revision = image_updater.next_revision
-
-      issues = Requirements::ImageRevisionChecker.new(next_image_revision)
-                                                 .pre_preview_metadata_issues
-
-      if issues.any?
-        flash.now["alert_with_items"] = {
-          "title" => I18n.t!("images.edit.flashes.requirements"),
-          "items" => issues.items,
-        }
-
-        render :edit,
-               assigns: { edition: edition, image_revision: next_image_revision, issues: issues },
-               layout: rendering_context,
-               status: :unprocessable_entity
-        next
-      end
-
-      updater = Versioning::RevisionUpdater.new(edition.revision, current_user)
-      updater.update_image(next_image_revision, params[:lead_image] == "on")
-
-      if updater.changed?
-        timeline_entry_type = if updater.selected_lead_image?
-                                :lead_image_selected
-                              elsif updater.removed_lead_image?
-                                :lead_image_removed
-                              else
-                                :image_updated
-                              end
-
-        TimelineEntry.create_for_revision(entry_type: timeline_entry_type, edition: edition)
-        edition.assign_revision(updater.next_revision, current_user).save!
-        PreviewService.new(edition).try_create_preview
-      end
-
-      if updater.selected_lead_image?
-        redirect_to document_path(edition.document),
+    result = Images::Update.call(params: params, user: current_user)
+
+    if result.failure?
+      flash.now["alert_with_items"] = {
+        "title" => I18n.t!("images.edit.flashes.requirements"),
+        "items" => result.issues.items,
+      }
+
+      render :edit,
+             assigns: { edition: result.edition,
+                        image_revision: result.image_revision,
+                        issues: result.issues },
+             layout: rendering_context,
+             status: :unprocessable_entity
+    else
+      document = result.edition.document
+
+      if result.updater.selected_lead_image?
```

Comment:
So for me I was thinking of updater as some of the key data that is part of the success - but it could always be assiged to context, particularly as we do same conditional in the interactor itself for the timeline entry.

Reply:
Yeah, I think that would be better. Otherwise, you have to be quite familiar with the code of the interactor; although this is the case for other everything else assigned to the result, at least for domain models, the value of the variable is pretty clear.

One annoyance I found when working on my own approach was the verbosity of using context/result everywhere, as well as the mystery of what's in the result. How would it be if we used something like `values_at` on result to get them all in one go?